src/components/Link/Link-v2.tsx

```diff
@@ -0,0 +1,96 @@
+import clsx from 'clsx';
+import React, { forwardRef } from 'react';
+import type { Size } from '../../util/variant-types';
+import type { IconName } from '../Icon';
+import Icon from '../Icon';
+
+import styles from './Link-v2.module.css';
+
+type LinkHTMLElementProps = Omit<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  'disabled'
+>;
+
+export type LinkProps = LinkHTMLElementProps & {
+  // Component API
+  /**
+   * Component used to render the element. Meant to support interaction with framework navigation libraries.
+   *
+   * **Default is `"a"`**.
+   */
+  as: string | React.ElementType;
```

Comment:
Adding this to support links link in Remix. Will likely need some work to support fully, but will be able to test once i get thru the first wave of component implementations

Reply:
Challenge I see: Remix links currently use `to` instead of `href`

So to enable passing in a Remix link to here, we'd probably need to figure out how to support passing the EDS Link's `href` prop to the Remix Link's `to` somehow.

Reply:
I want to replicate what we do [using `ClickableStyle`](https://github.com/chanzuckerberg/edu-stack/blob/main/app/components/Link.tsx) and also get rid of that component in the next version :). 

I didn't want to copy how `ClickableStyle` works in here, b/c it seems to lose all the typing information and converts everything to `any` :( What does make sense over there is that it uses a type param to combine the types of Remix's `Link` with the component props.

I'm guessing this was part of why you needed to export the types earlier?

Reply:
I agree on that goal, and not wanting to do ClickableStyle.

My point isn't that we shouldn't do this. It's that to achieve:

```ts
<EdsLink
  as={RemixLink}
  href={...} // or maybe to={...}?
/>
```

we'll need to figure out how to handle href vs to.

Reply:
> I agree on that goal, and not wanting to do ClickableStyle.
> 
> My point is that to achieve:
> 
> ```ts
> <EdsLink
>   as={RemixLink}
>   href={...} // or maybe to={...}?
> />
> ```
> 
> we'll need to figure out how to handle href vs to.

Yah, I think you misunderstood. The way `ClickableStyle` supports to/href is weird, but "works". So we are saying the same thing basically.

I want to avoid polluting EDS with anything specific to Remix or some other framework, as well. I like the TypeParam approach (e.g., combine `EDSLink` with Framework type, with option or to `Omit`). Once I get thru the re-theming, I'll try a few things.